metaci/build/management/commands/run_build.py

```diff
@@ -0,0 +1,74 @@
+import os
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand
+from django.db.models import signals
+
+from metaci.build import handlers
+from metaci.build.models import Build
+from metaci.build.tasks import lock_org, run_build, scratch_org_limits
+from metaci.cumulusci.models import Org
+from metaci.plan.models import Plan, PlanRepository
+from metaci.repository.models import Branch, Repository
+from metaci.utils import temp_disconnect_signal
+
+User = get_user_model()
+
+
+class Command(BaseCommand):
+    help = "Runs a build from the command line (on local computer)."
+
+    def add_arguments(self, parser):
+        parser.add_argument("repo_name", type=str)
+        parser.add_argument("branch_name", type=str)
+        parser.add_argument("plan_name", type=str)
+        parser.add_argument("commit", type=str)
+        parser.add_argument("username_or_email", type=str)
+
+    def handle(
+        self,
+        repo_name,
+        branch_name,
+        commit,
+        plan_name,
+        username_or_email,
+        *args,
+        **options,
+    ):
+        repo = Repository.objects.get(name=repo_name)
+        branch = Branch.objects.get(repo=repo, name=branch_name)
+        plan = Plan.objects.get(name=plan_name)
+        user = (
+            User.objects.filter(username=username_or_email).first()
+            or User.objects.filter(email=username_or_email).first()
+        )
+        assert user, (
+            "Could not find user with username or email matching " + username_or_email
+        )
+        planrepo = PlanRepository.objects.get(plan=plan, repo=repo)
+        build = Build.objects.create(
+            repo=repo,
+            plan=plan,
+            branch=branch,
+            planrepo=planrepo,
+            commit=commit,
+            build_type="manual",
+            user=user,
+            org=Org.objects.get(name=plan.org, repo=repo),
+        )
+        if build.org.scratch:
+            assert (
+                scratch_org_limits().remaining > settings.SCRATCH_ORG_RESERVE
+            ), "Not enough scratch orgs"
+        else:
+            status = lock_org(build.org, build.pk)
+
+            assert status, "Could not lock org"
+
+        dyno = os.environ["DYNO"]
+        print(f"Running build {build.pk} in {dyno}")
+        with temp_disconnect_signal(
+            signal=signals.post_save, receiver=handlers.queue_build, sender=Build
+        ):
```

Comment:
I really don't like this. For one thing I'm not sure it's threadsafe. Instead my suggestion would be to add a new option for build_type ("one-off"?) and make the signal handler skip queueing the build if it's that type.

Reply:
I can change it, but I think your concern is more aesthetic than practical. Keep in mind that we are talking about the last line of a command line command.

Reply:
I guess you're right that it's not a practical problem in this context, but yeah, I don't want to add code that's temporarily messing with signal handlers, and I think my suggestion is still a pretty straightforward alternative that avoids it.

Reply:
My preferred solution would be to get rid of the signal but I also don't want to take on that project right now.

Reply:
Yeah I agree with that as a longer term goal. Signals are for extensions/pluggability, not for assembling an application that we have full control over.

Reply:
Okay, it's changed now.